### ISSUE TITLE
[Block] Navigation nested inside [Block] Cover results in Z-Index issues with other cover blocks

### ISSUE THREAD

**Issue by cuemarie** [WordPress/gutenberg#49668]
### Description

If a navigation block is nested in a cover block, the navigation's submenu overlay overlaps with other Cover blocks on the page in the editor as well as in the live site.  This overlapping breaks the links of the overlay menu, and makes the submenu close when you try to hover over links.

### Step-by-step reproduction instructions

1. In an FSE theme, open the site editor
2. Insert a Cover Block, then nest a Navigation Block inside that Cover block
3. Add a 2nd Cover block after the first
4. Turn on the Navigation's Overlay Menu and test it in the editor and the live site
5. You can also see the same behavior with a submenu

### Screenshots, screen recording, code snippet

**Editor Behavior**

https://user-images.githubusercontent.com/27249804/230679982-3f1ebf81-cc5d-41a1-bf13-dbda8f938d05.mov

**Live Site Behavior**

https://user-images.githubusercontent.com/27249804/230680452-5f078034-8a6c-42f4-9b8b-7f1f07721bfa.mov

### Environment info

- Replicated in Wordpress 6.2 with Gutenberg deactivated
- Replicated with Gutenberg versions 15.4.0 and 15.5.0 active as well
- Theme used for testing: Twenty Twenty-Three

Originally reported here: https://github.com/Automattic/wp-calypso/issues/75439

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by kathrynwp** [comment]
I was able to replicate this with:

- WP 6.2
- Gutenberg 15.5.0  active or inactive
- No other plugins active
- Twenty Twenty-Three 1.1

**Comment by kathrynwp** [comment]
I think this may be a duplicate of #45353. @cuemarie, what do you think?

**Comment by cuemarie** [comment]
@kathrynwp yep, looks like the same symptoms in their recordings and report. Thanks for that find - I'll close this n favor of that one!

**Comment by entlein007** [comment]
I beg your pardon. I experienced an issue with a dropdown to a Navigation Block within a Cover Block on a live site. It displays underneath the following Main Section.
That behavior showed up when I updated to WP 6.3.1.  No Gutenberg-Plugin installed. I read in your post that a z-index issue pertaining to the Cover Block was already addressed (In Gutenberg plugin only or also in Core?) What do I need to do to use the fix? I went back to an earlier WP Version to get a quick fix to my problem in the meanwhile.  I am just a theme-user, first-timer on git-hub, not a developer, not sure how to address this.
